samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -233,6 +239,28 @@ void handleContainerLaunchFail(String processorId, String containerId, String pr
     }
   }
 
+  /**
+   * Handle the container stop failure for active containers and standby (if enabled).
+   * @param processorId logical id of the container eg 1,2,3
+   * @param containerId last known id of the container deployed
+   * @param containerHost host on which container is requested to be deployed
+   * @param containerAllocator allocator for requesting resources
+   */
+  void handleContainerStopFail(String processorId, String containerId, String containerHost,
+      ContainerAllocator containerAllocator) {
+    if (processorId != null && hasActiveContainerPlacementAction(processorId)) {
+      // Assuming resource acquired on destination host will be relinquished by the containerAllocator,
+      // We mark the placement action as failed, and return.
+      ContainerPlacementMetadata metaData = getPlacementActionMetadata(processorId).get();
```

Comment:
AMRMAsync thread will do a container status update here, allocator thread marks the actions as failed. Please remove the comments.